internal/impl/gcp/bigquery.go

```diff
@@ -0,0 +1,109 @@
+package gcp
```

Comment:
The code in this file has the dual purpose of making testing simpler by mocking the bigquery client and enabling the development of a similar `gcp_bigquery_select` processor (upcoming PR).